pandas/_libs/parsers.pyx

```diff
@@ -1238,9 +1260,21 @@ cdef class TextReader:
                                            na_filter, na_hashset, na_flist)
 
             if result is not None and dtype != 'float64':
-                result = result.astype(dtype)
+                if is_extension_array_dtype(dtype):
```

Comment:
why are you repeating?

Reply:
bad code... changing where the `is_extension_array_dtype` makes this unnecessary anyway.